### PR TITLE
Forcing reference kernel usage for TFLite interpreter during testing 

### PR DESCRIPTION
**Pull request by rewu93**
BUG=http://b/259572815

Currently, the default kernel usage of the TFlite interpreter is optimized ones. This creates test inconsistencies when comparing TFLM results (reference kernel) with TFL's. Inference results comparison is logically valid if both are using reference kernels.

### CODE REVIEW

File: tensorflow/lite/micro/python/interpreter/tests/interpreter_test.py [hunk 1]
```diff
@@ -143,7 +143,10 @@ def testCompareWithTFLite(self):
     tflm_interpreter = tflm_runtime.Interpreter.from_bytes(model_data)
 
     # TFLite interpreter
-    tflite_interpreter = tf.lite.Interpreter(model_content=model_data)
+    tflite_interpreter = tf.lite.Interpreter(
```
Comment (paulinesho) [inline tensorflow/lite/micro/python/interpreter/tests/interpreter_test.py:146]: Does removing tolerance in line 175 now work?

Comment (rewu93) [reply tensorflow/lite/micro/python/interpreter/tests/interpreter_test.py:146]: Yes. Removed. Again the real fix will be #1463

File: tensorflow/lite/micro/examples/hello_world/train/train_hello_world_model.py [hunk 2]
```diff
@@ -503,7 +503,9 @@ def predict_tflite(tflite_model: bytes, x_test: np.ndarray) -> np.ndarray:
   x_test_ = x_test_.astype(np.float32)
 
   # Initialize the TFLite interpreter
-  interpreter = tf.lite.Interpreter(model_content=tflite_model)
+  interpreter = tf.lite.Interpreter(model_content=tflite_model,
+                                    experimental_op_resolver_type=tf.lite.
+                                    experimental.OpResolverType.BUILTIN_REF)
```
Comment (advaitjain) [inline tensorflow/lite/micro/examples/hello_world/train/train_hello_world_model.py:508]: the formatting is odd (yapf does that sometimes).

Let's try adding a \:
```py
  interpreter = tf.lite.Interpreter(model_content=tflite_model,
                                    experimental_op_resolver_type=\
                                        tf.lite.experimental.OpResolverType.BUILTIN_REF)
```

Comment (advaitjain) [reply tensorflow/lite/micro/examples/hello_world/train/train_hello_world_model.py:508]: and if that works, perhaps make the same change in the other places as well?

Comment (rewu93) [reply tensorflow/lite/micro/examples/hello_world/train/train_hello_world_model.py]: Done

**Review by advaitjain — APPROVED** [review]

**Pull request merged**